asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractNodeImpl.java

```diff
@@ -54,157 +66,215 @@ public DocumentRuby document() {
 
     @Override
     public DocumentRuby getDocument() {
-        return this.abstractNode.getDocument();
+        return (DocumentRuby) NodeConverter.createASTNode(getProperty("document"));
```

Comment:
Yes! But I think the naming of the nodes is not final yet and Schalk has an immediate need for these changes.
So I'd like to see this coming first and I'll redo the class renaming once we've finished the renaming.